runtime/autoload/man.vim

```diff
@@ -11,127 +22,207 @@ catch /E145:/
+  let out = systemlist(s:man_cmd.s:man_find_arg.' '.s:man_args(sect, page))
+  if empty(out) || out[0] == ''
+    call s:error('no manual entry for '.page.(empty(sect)?'':'('.sect.')'))
+    return
+  elseif page !~# '\/' " if page is not a path, parse the page and section from the path
+    " use the last line because if we had something like printf(man) then man
+    " would be read as the manpage because man's path is at out[0]
+    let [page, sect] = s:parse_page_and_sect_path(out[len(out)-1])
+  endif
 
-  let [page, sect] = s:parse_page_and_section(sect, page)
+  call s:push_tag()
 
-  if !empty(sect) && s:find_page(sect, page) == 0
-    let sect = ''
-  endif
+  call s:read_page(sect, page, a:editcmd)
+endfunction
 
-  if s:find_page(sect, page) == 0
-    echo 'No manual entry for '.page
-    return
+" move to previous position in the stack
+function! man#pop_tag() abort
+  if !empty(s:tag_stack)
+    let tag = remove(s:tag_stack, -1)
+    execute tag['buf'].'b'
+    call cursor(tag['lnum'], tag['col'])
   endif
+endfunction
 
-  exec 'let s:man_tag_buf_'.s:man_tag_depth.' = '.bufnr('%')
-  exec 'let s:man_tag_lin_'.s:man_tag_depth.' = '.line('.')
-  exec 'let s:man_tag_col_'.s:man_tag_depth.' = '.col('.')
-  let s:man_tag_depth = s:man_tag_depth + 1
+" save current position in the stack
+function! s:push_tag() abort
+  let s:tag_stack += [{
+        \ 'buf':  bufnr('%'),
+        \ 'lnum': line('.'),
+        \ 'col':  col('.'),
+        \ }]
+endfunction
 
-  let editcmd = 'edit'
-  " Use an existing 'man' window, else open a new one.
-  if &filetype !=# 'man'
+" find the closest man window above/left
+function! s:find_man(cmd) abort
+  if g:man_find_window != 1 || &filetype ==# 'man'
+    return a:cmd
+  endif
+  if winnr('$') > 1
     let thiswin = winnr()
-    wincmd b
-    if winnr() > 1
-      exec thiswin . 'wincmd w'
-      while 1
-        if &filetype ==# 'man'
-          break
-        endif
-        wincmd w
-        if thiswin == winnr()
-          break
-        endif
-      endwhile
-    endif
+    while 1
+      if &filetype ==# 'man'
+        return 'edit'
+      endif
+      wincmd w
+      if thiswin == winnr()
+        return a:cmd
+      endif
+    endwhile
+  endif
+  return a:cmd
+endfunction
 
-    if &filetype !=# 'man'
-      let editcmd = 'tabnew'
-    endif
+" parses the page and sect out of 'page(sect)'
+function! s:parse_page_and_sect_fpage(fpage) abort
+  let ret = split(a:fpage, '(')
+  if len(ret) > 1
+    let iret = split(ret[1], ')')
+    return [ret[0], tolower(iret[0])]
+  else
+    return [ret[0], '']
   endif
+endfunction
 
-  silent exec editcmd.' man://'.page.(empty(sect)?'':'('.sect.')')
+" parses the page and sect out of 'path/page.sect'
+function! s:parse_page_and_sect_path(path) abort
+  let tail = fnamemodify(a:path, ':t')
+  if fnamemodify(tail, ':e') =~# s:man_extensions
+    let tail = fnamemodify(tail, ':r')
+  endif
+  " TODO(nhooyr) all substitutes to matchlist/matchstr maybe?
```

Comment:
Since you're only capturing one group, take a look at `matchstr()`.  `matchstr(tail, '\f\+')` should be good enough for `page`.  You can `\zs` and to extract the the text for `sect`.